### PR TITLE
Multiple shipping addresses PR

### LINKED ISSUE #1693: Multiple shipping addresses
Add support for our new backend feature.

### PR DESCRIPTION
**Pull request by acbeni**
# Description

Closes #1693

# Checklist

- [x] Update release Notes
- [x] Add to the current github milestone

### CODE REVIEW

File: commercetools-models/src/main/java/io/sphere/sdk/models/Address.java [hunk 1]
```diff
@@ -33,13 +33,15 @@
  * @see io.sphere.sdk.carts.commands.updateactions.SetBillingAddress
  * @see io.sphere.sdk.carts.commands.updateactions.SetShippingAddress
  */
-public final class Address extends Base {
+public final class Address extends Base implements WithKey{
```
Comment (katmatt) [inline commercetools-models/src/main/java/io/sphere/sdk/models/Address.java]: Great that you added the `WithKey` here 👍

File: commercetools-models/src/main/java/io/sphere/sdk/orders/LineItemImportDraftBuilder.java [hunk 2]
```diff
@@ -65,12 +68,16 @@ public LineItemImportDraftBuilder custom(@Nullable final CustomFieldsDraft custo
         return this;
     }
 
+    public LineItemImportDraftBuilder custom(@Nullable final ItemShippingDetailsDraft shippingDetails) {
```
Comment (katmatt) [inline commercetools-models/src/main/java/io/sphere/sdk/orders/LineItemImportDraftBuilder.java]: Shouldn't this builder method be called `shippingDetails`?

File: commercetools-models/src/main/java/io/sphere/sdk/orders/OrderImportDraftImpl.java [hunk 3]
```diff
@@ -198,4 +201,10 @@ public TaxCalculationMode getTaxCalculationMode() {
     public CartOrigin getOrigin() {
         return origin;
     }
+
+    @Nullable
+    @Override
+    public List<Address> getItemShippingAddresses() {
+        return null;
```
Comment (katmatt) [inline commercetools-models/src/main/java/io/sphere/sdk/orders/OrderImportDraftImpl.java]: Why do we return `null` here?

File: commercetools-models/src/main/java/io/sphere/sdk/shippingmethods/errors/InvalidItemShippingDetailsError.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        this.subject = subject;
+        this.itemId = itemId;
+    }
+
+    public static InvalidItemShippingDetailsError of(final String message, final String subject, final String itemId) {
+        return new InvalidItemShippingDetailsError(message, subject, itemId);
+    }
+
+    /**
+     * @return Either {@code LineItem} or {@code CustomLineItem} to mark if it is about a line item or a custom line item.
```
Comment (katmatt) [inline commercetools-models/src/main/java/io/sphere/sdk/shippingmethods/errors/InvalidItemShippingDetailsError.java]: It took a moment for me to understand that these are string constants, may be writing it as `{@code"LineItem"} would have helped? What do you think?

File: commercetools-models/src/test/java/io/sphere/sdk/carts/MultipleShippingAddressesIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+import static io.sphere.sdk.carts.CartFixtures.withCartDraft;
+import static io.sphere.sdk.carts.CartFixtures.withFilledCart;
+import static io.sphere.sdk.products.ProductFixtures.withTaxedProduct;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class MultipleShippingAddressesIntegrationTest extends IntegrationTest {
```
Comment (katmatt) [inline commercetools-models/src/test/java/io/sphere/sdk/carts/MultipleShippingAddressesIntegrationTest.java]: I really like that you create separate integration test for each new feature 👍

File: commercetools-models/src/test/java/io/sphere/sdk/products/ProductFixtures.java [hunk 6]
```diff
@@ -107,8 +107,8 @@ public static void withUpdateableProduct(final BlockingSphereClient client, fina
         withUpdateableProduct(client, randomString(), f);
     }
 
-    public static void withProduct(final BlockingSphereClient client, final Consumer<Product> user) {
-        withProduct(client, randomString(), user);
+    public static void withProduct(final BlockingSphereClient client, final Consumer<Product> productConsumer) {
```
Comment (katmatt) [inline commercetools-models/src/test/java/io/sphere/sdk/products/ProductFixtures.java]: Good catch 👍

File: commercetools-models/src/main/java/io/sphere/sdk/orders/LineItemImportDraftBuilder.java (hunk 2 above)
Comment (acbeni) [reply commercetools-models/src/main/java/io/sphere/sdk/orders/LineItemImportDraftBuilder.java]: 😳  copy past mistake 😞

File: commercetools-models/src/main/java/io/sphere/sdk/orders/OrderImportDraftImpl.java (hunk 3 above)
Comment (acbeni) [reply commercetools-models/src/main/java/io/sphere/sdk/orders/OrderImportDraftImpl.java]: 👍  nice catch

File: commercetools-models/src/test/java/io/sphere/sdk/carts/MultipleShippingAddressesIntegrationTest.java (hunk 5 above)
Comment (acbeni) [reply commercetools-models/src/test/java/io/sphere/sdk/carts/MultipleShippingAddressesIntegrationTest.java]: Yep it looks like a ref 👍 
Now fixed

**Comment by acbeni** [comment]
@katmatt The code review has been taken care of.

**Pull request merged**